### PR TITLE
Milestone 2 : Use the "Decorator" design pattern.

### PR DESCRIPTION
**Pull request by ABDULALI3468**
> ## What I did ♻️
> > In this project's milestone I use the `Decorator` design pattern to correct the names of people.
> 
> ## Project requirements
> * Check the usage of [Decorator pattern in Ruby](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0) ✅
> * Think about how you can use two decorators in order to capitalize and trim people's names.
> * In the following steps you will get instructions for implementing the Decorator design pattern for this case.
> 
> #### Interface ✅
> * Create a class `Nameable`.
>   
>   * Implement a method called `correct_name` that will raise a `NotImplementedError`.
> 
> #### Turn your Person class to Nameable ✅
> * Make sure that your Person class inherits from Nameable
> * Make sure that this class has a method `correct_name` implemented. It should simply return the name attribute.
> 
> #### Prepare base Decorator ✅
> * Make sure that it inherits from Nameable.
> * In the constructor assign a nameable object from params to an instance variable.
> * Implement the `correct_name` method that returns the result of the correct_name method of the `@nameable`.
> 
> #### Prepare CapitalizeDecorator and TrimmerDecorator ✅
> * For the CapitalizeDecorator:
>   
>   * Create a class that inherits from the base Decorator class.
>   * Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
> * For the TrimmerDecorator:
>   
>   * Create a class that inherits from the base Decorator class.
>   * Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.
> 
> ### See your decorators in action ✅
> \Try the following code and check if you managed to decorate your person:
> 
> ```ruby
> person = Person.new(22, 'maximilianus')
>   person.correct_name
>   capitalizedPerson = CapitalizeDecorator.new(person)
>   capitalizedPerson.correct_name
>   capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
>   capitalizedTrimmedPerson.correct_name
> ```

### CODE REVIEW

File: decorators.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
```
Comment (BenMukebo) [inline decorators.rb:5]: - [OPTIONAL] Let's raise an error message to the method 'correct_name'. It should look something like the following:

```ruby
class Nameable
    def correct_name
        raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
    end
end
```

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  private :of_age?
+end
+
+person = Person.new(22, 'maximilianus')
+person.correct_name
+capitalized_person = CapitalizeDecorator.new(person)
+puts capitalized_person.correct_name
+capitalized_trimmed_person = TrimmerDecorator.new(capitalizedPerson)
```
Comment (BenMukebo) [inline person.rb:33]: - [OPTIONAL] There is a typo on line 33, the variable or method 'capitalizedPerson' does not exist this should be `capitalized_person` instead:
  -  ```capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)```
- Kindly correct this 👍🏻

**Review by BenMukebo — APPROVED** [review]
<h1>Status: Approve  :champagne:  :+1: :confetti_ball: 

### **Hi @ABDULALI3468** 

### **Congratulations! 🎉 Project Approved ✅**

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 


### To Highlight :1st_place_medal: 
- Descriptive README. ✔️  
- No Linter errors :heavy_check_mark: 
- You managed to decorate a class 🥇 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (ABDULALI3468) [reply person.rb:33]: @BenMukebo thank you so much I will correct them soon. Have a nice life.
Kind regards
@ABDULALI3468

Comment (BenMukebo) [reply person.rb:33]: Thank you!!

**Pull request merged**